app/src/util/ingest_utils.py

```diff
@@ -58,3 +59,18 @@ def process_and_ingest_sys_args(argv: list[str], logger: Logger, ingestion_call:
         db_session.commit()
 
     logger.info("Finished processing")
+
+
+def tokenize(text: str) -> list[str]:
+    """
+    The add_special_tokens argument is specified in PreTrainedTokenizerFast.encode_plus(), parent class of MPNetTokenizerFast.
+    It defaults to True for encode_plus() but defaults to False for .tokenize().
+    Setting add_special_tokens=True will add the special tokens CLS(0) and SEP(2) to the beginning and end of the input text.
+    """
+    tokenizer = app_config.sentence_transformer.tokenizer
+    if isinstance(tokenizer, PreTrainedTokenizerFast):
+        return tokenizer.tokenize(text, add_special_tokens=True)
+    elif tokenizer.__class__.__name__ == "MockTokenizer":
+        return tokenizer.tokenize(text)
```

Comment:
Suggest instead we modify the mock to accept the `add_special_tokens=True` argument rather than detect it in the prod code

Reply:
Does the MockTokenizer have/use special tokens?

Reply:
We can make it behave that way -- that's better than trying to catch the class instance here.

Reply:
It's not clear to me how it's better but I'll update MockTokenizer.

Reply:
> it gives future maintainers the appearance of type safety

I wouldn't interpret that the code is typesafe, but rather that the code only works for those explicit types.

Reply:
> I wouldn't interpret that the code is typesafe, but rather that the code only works for those explicit types.

Right, but we should try to _make_ it type safe: if someone writes buggy code that passes in a variable not of those types, it's better for that code to fail a linter check than to fail at runtime.

Reply:
In other words, if the intent of the code is "this only works for these two types," that should be communicated through the type system, which will enforce that intent before code gets merged, than through logical code, which won't fail until runtime.